src/Bitcoin/TransactionSigner.cpp

```diff
@@ -53,6 +59,8 @@ Result<Transaction> TransactionSigner<Transaction, TransactionBuilder>::sign() {
 template <typename Transaction, typename TransactionBuilder>
 Result<void> TransactionSigner<Transaction, TransactionBuilder>::sign(Script script, size_t index,
                                                   const Bitcoin::Proto::UnspentTransaction& utxo) {
+    assert(index < transaction.inputs.size());
```

Comment:
How we can catch this in JNI?

Reply:
Product is (should) be compiled in release mode, when asserts do nothing.  But it is a good question what happens in the JNI wrapper if there is an assert in the C++ code below; I should create a Kotlin test for it.  But this is not really possible, because the combinations that would lead to assert are filtered out earlier (like there are 0 utxos).

Reply:
App just crushed.

Reply:
@madcake how will it break android? assert will be removed in released lib

Reply:
> App just crushed.

@madcake, can you share the exact input with which you produced crash?